### PR TITLE
Add mode documentation and example in README

### LINKED ISSUE #7: Improve README
- [x] Add more examples
- [x] Testing section

### PR DESCRIPTION
**Pull request by nicolas-brousse**
Closes #7.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,7 +119,11 @@ decorated_user # => UserDecorator
 decorated_user.posts.first # => PostDecorator
 ```
 
-### Specify decorator
+### Custom decorator
+
+By default dekorator search the decorator class by adding `Decorator` at the end.
```
Comment (Spone) [inline README.md]:
```suggestion
By default, Dekorator searches for the decorator class by adding `Decorator` at the end.  
```

File: README.md [hunk 2]
```diff
@@ -95,7 +119,11 @@ decorated_user # => UserDecorator
 decorated_user.posts.first # => PostDecorator
 ```
 
-### Specify decorator
+### Custom decorator
+
+By default dekorator search the decorator class by adding `Decorator` at the end.  
+For `User`, dekorator looks for `UserDecorator` class. And so for `User::Profile`
```
Comment (Spone) [inline README.md]:
```suggestion
For `User`, Dekorator looks for the `UserDecorator` class, and for `User::Profile`
```

File: README.md [hunk 3]
```diff
@@ -165,6 +193,26 @@ class UserDecorator < ApplicationDecorator
 end
 ```
 
+## Testing
+
+`rails generate dekorator:decorator user` also generate a testing file based on your
```
Comment (Spone) [inline README.md]:
```suggestion
`rails generate dekorator:decorator user` also generates a testing file based on your
```

**Comment by nicolas-brousse** [comment]
I'll wait to #32 to be closed before continuing this.

File: README.md [hunk 4]
```diff
@@ -165,6 +193,26 @@ class UserDecorator < ApplicationDecorator
 end
 ```
 
+## Testing
+
+`rails generate decorator user` also generate a testing file based on your
```
Comment (Spone) [inline README.md]:
```suggestion
`rails generate decorator user` also generates a testing file based on your
```

**Pull request merged**